src/realm/dictionary.cpp

```diff
@@ -740,6 +738,9 @@ void Dictionary::clear()
         if (recurse)
             _impl::TableFriend::remove_recursive(*m_obj.get_table(), cascade_state); // Throws
     }
+    Replication* repl = m_obj.get_replication();
```

Comment:
You will need to call the replication before doing the actual clear. Otherwise you will not get the correct prior size.

Reply:
done